src/openrct2/world/mapgen.c

```diff
@@ -764,3 +782,247 @@ static void mapgen_simplex(mapgen_settings *settings)
 }
 
 #pragma endregion
+
+#pragma region Heightmap
+
+bool mapgen_load_heightmap(const utf8 *path)
+{
+	const char* extension = path_get_extension(path);
+	uint8 *pixels;
+	size_t pitch;
+	uint32 numChannels;
+	uint32 width, height;
+
+	if (strcicmp(extension, ".png") == 0) {
+		if (!image_io_png_read(&pixels, &width, &height, path)) {
+			log_warning("Error reading PNG");
+			window_error_open(STR_HEIGHT_MAP_ERROR, STR_ERROR_READING_PNG);
+			return false;
+		}
+
+		numChannels = 4;
+		pitch = width * numChannels;
+	}
+	else if (strcicmp(extension, ".bmp") == 0) {
+		SDL_Surface *bitmap = SDL_LoadBMP(path);
```

Comment:
@janisozaur Why are you pinging me?

If its only available via the UI then it won't be a problem anyway.